tests/test_subtractTask.py

```diff
@@ -715,6 +715,76 @@ def _compare_apCorrMaps(self, a, b):
+        )
+        template, _ = makeTestImage(psfSize=2.4, xSize=xSize, ySize=ySize, doApplyCalibration=True)
+        tmplt_fake_img, tmplt_fake_sources = makeTestImage(
+            psfSize=2.4, xSize=xSize, ySize=ySize, seed=9, nSrc=2, noiseLevel=0.25, fluxRange=1
+        )
+        # created fakes and added them to the images
+        science.image.array += science_fake_img.image.array
+        template.image.array += tmplt_fake_img.image.array
+
+        # TODO: update to INJECTED names when source injection gets refactored
+        # adding mask planes to both science and template images
+        science.mask.addMaskPlane("FAKE")
+        template.mask.addMaskPlane("FAKE")
+
+        science_mask_planes = science.mask.getMaskPlaneDict()
+        template_mask_planes = template.mask.getMaskPlaneDict()
+
+        x0 = science.mask.getX0()
+        y0 = science.mask.getY0()
+        for a_science_source in science_fake_sources:
+            # 3 x 3 masking of the source locations is fine
+            for dy in (-1, 0, 1):
+                science.mask.setMaskPlaneValues(
+                    science_mask_planes['FAKE'],
+                    int(a_science_source.getX() - x0)-1,
+                    int(a_science_source.getX() - x0)+1,
+                    int(a_science_source.getY() - y0)+dy
+                )
+
+        x0 = template.mask.getX0()
+        y0 = template.mask.getY0()
+        for a_template_source in tmplt_fake_sources:
+            # 3 x 3 masking of the source locations is fine
+            for dy in (-1, 0, 1):
+                template.mask.setMaskPlaneValues(
+                    template_mask_planes['FAKE'],
+                    int(a_template_source.getX() - x0)-1,
+                    int(a_template_source.getX() - x0)+1,
+                    int(a_template_source.getY() - y0)+dy
+                )
+
+        science_fake_masked = (science.mask.array & science.mask.getPlaneBitMask("FAKE")) > 0
+        template_fake_masked = (template.mask.array & template.mask.getPlaneBitMask("FAKE")) > 0
+
+        config = subtractImages.AlardLuptonSubtractTask.ConfigClass()
+        config.mode = "convolveTemplate"
+        task = subtractImages.AlardLuptonSubtractTask(config=config)
+        subtraction = task.run(template, science, sources)
+
+        # check subtraction mask plane is set where we set the previous masks
+        diff_mask = subtraction.difference.mask
+
+        # science mask should be now in INJECTED
+        inj_masked = (diff_mask.array & diff_mask.getPlaneBitMask("INJECTED")) > 0
+
+        # template mask should be now in INJECTED_TEMPLATE
+        injTmplt_masked = (diff_mask.array & diff_mask.getPlaneBitMask("INJECTED_TEMPLATE")) > 0
+
+        self.assertEqual(np.sum(inj_masked.astype(int)-science_fake_masked.astype(int)), 0)
+        self.assertEqual(np.sum(injTmplt_masked.astype(int)-template_fake_masked.astype(int)), 0)
```

Comment:
Since the mask plane is changing, this looks like a good way to do the test. For future reference, see https://pipelines.lsst.io/py-api/lsst.utils.tests.TestCase.html for useful built-in asserts, though I don't think any would quite work here.